src/libfuncs/cast.rs

```diff
@@ -182,14 +159,11 @@ pub fn build_downcast<'ctx, 'this>(
         };
 
         let result = if src_width > dst_width {
-            block
-                .append_op_result(arith::trunci(src_value, dst_ty, location))?
+            block.append_operation(arith::trunci(src_value, dst_ty, location))
```

Comment:
note that here your change was correct (now it isnt), because the result is actually used (its a if block returning the result)